specs/ExecuteScript.md

```diff
@@ -0,0 +1,222 @@
+  /// `NavigationStarting` event handlers. All of the event handlers share a
+  /// common `NavigationStartingEventArgs` object. Whichever event handler is
+  /// last to change the `NavigationStartingEventArgs.Cancel` property will
+  /// decide if the frame navigation will be cancelled. Redirects raise this
+  /// event as well, and the navigation id is the same as the original one.
+  ///
+  /// You may block corresponding navigations until the event handler returns.
+  ///
+  /// \snippet ScenarioWebViewEventMonitor.cpp NavigationStarting
+  HRESULT add_NavigationStarting(
+      [in] ICoreWebView2FrameNavigationStartingEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_NavigationStarting`.
+  HRESULT remove_NavigationStarting(
+      [in] EventRegistrationToken token);
+
+  /// Add an event handler for the `ContentLoading` event.  `ContentLoading`
+  /// triggers before any content is loaded, including scripts added with
+  /// `AddScriptToExecuteOnDocumentCreated`.  `ContentLoading` does not trigger
+  /// if a same page navigation occurs (such as through `fragment`
+  /// navigations or `history.pushState` navigations).  This operation
+  /// follows the `NavigationStarting` and precedes `NavigationCompleted` events.
+  HRESULT add_ContentLoading(
+      [in] ICoreWebView2FrameContentLoadingEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_ContentLoading`.
+  HRESULT remove_ContentLoading(
+      [in] EventRegistrationToken token);
+
+  /// Add an event handler for the `NavigationCompleted` event.
+  /// `NavigationCompleted` runs when the CoreWebView2Frame has completely
+  /// loaded (`body.onload` runs) or loading stopped with error.
+  ///
+  /// \snippet ScenarioWebViewEventMonitor.cpp NavigationCompleted
+  HRESULT add_NavigationCompleted(
+      [in] ICoreWebView2FrameNavigationCompletedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_NavigationCompleted`.
+  HRESULT remove_NavigationCompleted(
+      [in] EventRegistrationToken token);
+
+  /// Add an event handler for the DOMContentLoaded event.
+  /// DOMContentLoaded is raised when the iframe html document has been parsed.
+  /// This aligns with the the document's DOMContentLoaded event in html.
+  ///
+  /// \snippet ScenarioWebViewEventMonitor.cpp DOMContentLoaded
+  HRESULT add_DOMContentLoaded(
+      [in] ICoreWebView2FrameDOMContentLoadedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove an event handler previously added with add_DOMContentLoaded.
+  HRESULT remove_DOMContentLoaded(
+      [in] EventRegistrationToken token);
+
+  /// Run JavaScript code from the javascript parameter in the current frame.
+  /// The result of evaluating the provided JavaScript is used in this parameter.
```

Comment:
```suggestion
  /// The result of evaluating the provided JavaScript is passed to the completion handler.
```

Reply:
Please update and also update in the ICoreWebView2::ExecuteScript documentation (thx!)